### PR TITLE
Surface information about a location to the dataLayer

### PR DESCRIPTION
**Pull request by andrewgarner**
Surfaces data about locations to the Google Tag Manager dataLayer.

### CODE REVIEW

File: features/step_definitions/location_steps.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@
   location.name.click
 end
 
-When(/^I view the details of an appointment location that handles its own booking$/) do
+When(/^(?:I view|a user views) the details of an appointment location(?: that handles its own booking)?$/) do
```
Comment (stevenwilkin) [inline features/step_definitions/location_steps.rb]: Perhaps define a separate `When` instead of introducing complexity into the regex here?

**Comment by stevenwilkin** [comment]
Any reason for testing this with Cucumber?

This feature isn't adding any value to the end user and my concern is that it's bloating the set of acceptance criteria for the app

File: features/support/matchers/have_data_layer_information.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  def wait_until(timeout_in_seconds = Capybara.default_wait_time)
+    Timeout.timeout(timeout_in_seconds) do
+      sleep(0.1) until (value = yield)
+      value
+    end
+  end
+
+  match do |page|
+    wait_until do
+      @gaq = Array(page.evaluate_script('window.dataLayer'))
```
Comment (stevenwilkin) [inline features/support/matchers/have_data_layer_information.rb]: Does this need to be an instance variable?

**Comment by stevenwilkin** [comment]
Can the location id not be deduced from the page url in GA?

**Comment by andrewgarner** [comment]
I started off with a request spec, but, Capybara is no longer supported in
request specs as of Capybara 2.0.0. The recommended way to use Capybara is
with feature specs.

Given we have Cucumber features I decided to follow the same path. I also think there is value in documenting the need of an analyst with gherkins. I'd argue they are an end-user of this application.

**Comment by stevenwilkin** [comment]
Using Gherkin to assert that a specific piece of JavaScript has been run just seems wrong to me. If this *has* to be done I'd much rather see it done with RSpec.

**Comment by andrewgarner** [comment]
cc/ @guidance-guarantee-programme/developers - now testing with RSpec.

**Comment by stevenwilkin** [comment]
03485a74b74fd690449136fa0d06e97bcca5f6c1 refactors a lot of the postcode stuff in the controller along with the dataLayer piece. Would be easier to follow if this was 2 commits.

**Comment by stevenwilkin** [comment]
VCR cassettes could be better in their own commit

**Comment by andrewgarner** [comment]
In regards to committing the cassettes separately – shouldn't aspire for each commit to be green?

**Comment by stevenwilkin** [comment]
It's an ideological issues I guess, I just find it easier to review code when changes to cassettes are segregated.

**Pull request merged**